lib/core/src/server/preview/virtualModuleEntry.template.js

```diff
@@ -1,29 +1,32 @@
 /* eslint-disable import/no-unresolved */
 import { addDecorator, addParameters, addLoader, addArgTypesEnhancer } from '{{clientApi}}';
 import { logger } from '{{clientLogger}}';
-import {
-  decorators,
-  parameters,
-  loaders,
-  argTypesEnhancers,
-  globals,
-  globalTypes,
-  args,
-  argTypes,
-} from '{{configFilename}}';
+import * as config from '{{configFilename}}';
 
-if (args || argTypes) {
-  logger.warn('Invalid args/argTypes in config, ignoring.', JSON.stringify({ args, argTypes }));
-}
-if (decorators) {
-  decorators.forEach((decorator) => addDecorator(decorator, false));
-}
-if (loaders) {
-  loaders.forEach((loader) => addLoader(loader, false));
-}
-if (parameters || globals || globalTypes) {
-  addParameters({ ...parameters, globals, globalTypes }, false);
-}
-if (argTypesEnhancers) {
-  argTypesEnhancers.forEach((enhancer) => addArgTypesEnhancer(enhancer));
-}
+Object.entries(config).forEach(([key, value]) => {
+  switch (key) {
+    case 'args':
+    case 'argTypes': {
+      return logger.warn('Invalid args/argTypes in config, ignoring.', JSON.stringify(value));
+    }
+    case 'decorators': {
+      return value.forEach((decorator) => addDecorator(decorator, false));
+    }
+    case 'loaders': {
+      return value.forEach((loader) => addLoader(loader, false));
+    }
+    case 'parameters': {
+      return addParameters({ ...value }, false);
+    }
+    case 'argTypesEnhancers': {
+      return value.forEach((enhancer) => addArgTypesEnhancer(enhancer));
+    }
+    case 'globals':
+    case 'globalTypes': {
+      return addParameters({ [key]: value }, false);
+    }
+    default: {
+      return console.log(`${key} was not supported`);
+    }
+  }
+});
```

Comment:
by iterating over the module's exports we don't check for the existence of any don't don't.

Therefor we avoid the `export '(decorators|parameters|argTypesEnhancers|globals|globalTypes|args|argTypes)' was not found in` warning

Reply:
Nice @ndelangen I knew you could figure it out ;)

Reply:
I had to lower this to ES5, for IE11